shared/macros-oval.jinja

```diff
@@ -0,0 +1,149 @@
+{{%- endif %}}
+      test_ref="test_{{{ rule_id }}}{{{ none_exist }}}" />
+{{%- endmacro %}}
+
+{{#
+  Macro to define the OVAL object to be constructed (Test definition).
+  This function can take three values:
+    - path (String): Path to the configuration file to be checked.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - none_exist (Any String): If set, the expanded OVAL check will check for the absence of a parameter in a configuration file.
+#}}
+{{%- macro oval_line_in_file_test(path='', parameter='', none_exist='') -%}}
+  {{%- if none_exist %}}
+  <ind:textfilecontent54_test check="all" check_existence="none_exist"
+  comment="tests the absence of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- else %}}
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="tests the value of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- endif %}}
+  id="test_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+  <ind:object object_ref="obj_{{{ rule_id }}}{{{ none_exist }}}" />
+  {{%- if not none_exist %}}
+  <ind:state state_ref="state_{{{ rule_id }}}" />
+  {{%- endif %}}
+  </ind:textfilecontent54_test>
+{{%- endmacro %}}
+
+{{#
+  Macro to check if a parameter in a configuration file is set (Object definition).
+  This function can take five values:
+    - path (String): Path to the configuration file to be checked.
+    - prefix_regex (String): Regular expression to be used in the beginning of the OVAL text file content check.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - separator_regex (String): Regular expression to be used as the separator of parameter and value in a configuration file.
+    - none_exist (Any String): If set, the expanded OVAL check will check for the absence of a parameter in a configuration file.
+#}}
+{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', none_exist='') -%}}
+{{%- set regex = prefix_regex+parameter+separator_regex -%}}
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+    <ind:filepath>{{{ path }}}</ind:filepath>
+    {{%- if none_exist %}}
+    <ind:pattern operation="pattern match">{{{ regex }}}</ind:pattern>
+    {{%- else %}}
+    <ind:pattern operation="pattern match">{{{ regex }}}(.*)[\s]*(?:|(?:#.*))?$</ind:pattern>
+    {{%- endif %}}
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+{{%- endmacro %}}
+
+{{#
+  Macro to check if a expected value can be found in the extracted information of an OVAL object (State definition).
+  This function can take twi values:
+    - value (String): The value to be checked. This can also be a regular expression (e.g: value1|value2 can match both values).
+    - multi_value (Any String): If set, multiple values can be present for the parameter (e.g: parameter="value1 value2 value3").
```

Comment:
This description is a bit confusing. Meaning is if you set `multi_value` then you can use multiple values in `value` parameter, right? 
e.g. `multi_value (Any String): If set, multiple values can be present for parameter "value" (e.g: value="value1 value2 value3").` would be more accurate.

Reply:
Maybe the concept (`multi_value`) is still confusing... I'll try to come up with something more clear.

Reply:
It's much better. Only one thing - maybe add (to brackets or after `the parameter` word), that with phrase `the parameter` you mean parameter `value` (not parameter `multi_value`). Because user may get from the documentation, that if he wants multiple values, then he should input them in `multi_value`.

Reply:
This version is outdated. I rephrased this sentence to:

`multi_value (boolean): If set, it means that the parameter can accept multiple values and the expected value must be present in the current list of values.`